geomstats/visualization/grassmannian.py

```diff
@@ -0,0 +1,734 @@
+        n = self.n  # Size of space
+        k = self.k  # Subspace size
+        if k > 1 or n > 3:
+            print(
+                "Cannot handle sizes of k larger than 1 at this time.\
+                 Cannot handle n larger than 3 at this time."
+            )
+            return
+
+        # Plotting for 2d
+        if n == 2:
+            projections = self.random_uniform(100)
+            self.plot(ticks)
+            plot_me = projection_to_two_d(projections, np.shape(projections)[0])
+            for i in range(np.shape(plot_me)[0]):
+                plt.plot(plot_me[i][0], plot_me[i][1], "y.")
+
+        if n == 3:
+            points = 10000
+
+            point_draw = 100
+
+            fig = plt.figure()
+            ax = fig.add_subplot(projection="3d")
+            u, v = np.mgrid[0 : 2 * np.pi : 30j, 0 : np.pi : 20j]
+            x = np.cos(u) * np.sin(v)
+            y = np.sin(u) * np.sin(v)
+            z = np.cos(v)
+            if ticks is True:
+                ax.plot_wireframe(x, y, abs(z), color="magenta", alpha=0.1)
+
+            bunch_of_projections = self.random_uniform(points)
+            bunch_two = self.random_uniform(point_draw)
+
+            # plot manifold
+            for i in range(points):
+                vector = np.random.rand(3, 1)
+                x, y, z = np.matmul(bunch_of_projections[i], vector) / np.linalg.norm(
+                    np.matmul(bunch_of_projections[i], vector)
+                )
+                ax.plot(x, y, abs(z), "sk")
+
+            # plot points on manifold
+            for i in range(point_draw):
+                vector = np.random.rand(3, 1)
+                x, y, z = np.matmul(bunch_two[i], vector) / np.linalg.norm(
+                    np.matmul(bunch_two[i], vector)
+                )
+                ax.plot(x, y, abs(z), "sm", alpha=0.3)
+
+            ax.set_xlim([-1, 1])
+            ax.set_ylim([-1, 1])
+            ax.set_zlim([0, 1])
+            ax.view_init(10, 80)
+            # Hide grid lines
+            ax.grid(False)
+            # Hide axes ticks
+            ax.set_xticks([])
+            ax.set_yticks([])
+            ax.set_zticks([])
+            plt.show()
+
+    def plot_tangent_space(self):
+        """Plot Grassmanian tangent space.
+
+        Warning: Under construction, do not use.
+        Only functions in 2D in limited cases.
+        """
+        self.plot(True)
+        projection = self.random_uniform(1)
+        point = projection_to_two_d(projection, 1)
+
+        print("Random point is: ", point[0], " ", point[1])
+
+        plt.plot(point[0], point[1], "g+")
+
+        normal_circle = [0, 0, 1]
+
+        point_in_3d = [point[0], point[1], 0]
+
+        tangent = np.cross(point_in_3d, normal_circle)
+
+        x_part = [point[0], point[0] + tangent[0]]
+
+        y_part = [point[1], point[1] + tangent[1]]
+
+        plt.plot(x_part, y_part, "-g")
+
+
+class GrassmannianCanonicalMetric(MatricesMetric, RiemannianMetric):
```

Comment:
Remove this class entirely. If you need the metric, import it direclty from geomstats:

`from geomstats.geometry.grassmannian import GrassmannianCanonicalMetric`